src/sparse/KokkosSparse_CrsMatrix.hpp

```diff
@@ -519,27 +519,21 @@ class CrsMatrix {
   /// \param nrows [in] The number of rows.
   /// \param ncols [in] The number of columns.
   /// \param annz [in] The number of entries.
-  /// \param val [in] The entries.
-  /// \param rows [in] The row indices.  rows[k] is the row index of
-  ///   val[k].
-  /// \param cols [in] The column indices.  cols[k] is the column
-  ///   index of val[k].
-  /// \param pad [in] If true, pad the sparse matrix's storage with
-  ///   zeros in order to improve cache alignment and / or
-  ///   vectorization.
-  ///
-  /// FIXME (mfh 21 Jun 2013) The \c pad argument is currently not used.
+  /// \param val [in] The values.
+  /// \param rowmap [in] The row offsets. The values/columns in row k begin at index
+  ///   \c rowmap[k] and end at \c rowmap[k+1]-1 (inclusive). This means the array
+  ///   must have length \c nrows+1.
+  /// \param cols [in] The column indices. \c cols[k] is the column
+  ///   index of entry k, with a corresponding value of \c val[k] .
   CrsMatrix (const std::string &label,
              OrdinalType nrows,
              OrdinalType ncols,
              size_type annz,
              ScalarType* val,
-             OrdinalType* rows,
-             OrdinalType* cols,
-             bool pad = false)
+             OrdinalType* rowmap,
+             OrdinalType* cols)
```

Comment:
You are sure Tpetra doesn't call this constructor w/ pads set ?

Reply:
@srajama1 I'll find out. But the pad argument never had any effect, so I doubt it. We can also catch and fix that easily during promotion if anyone did pass in something.